osgeo_importer/views.py

```diff
@@ -119,11 +111,12 @@ def form_valid(self, form):
                 for layer in description:
                     configuration_options = DEFAULT_LAYER_CONFIGURATION.copy()
                     configuration_options.update({'index': layer.get('index')})
-                    upload.uploadlayer_set.add(UploadLayer(name=upfile_basename,
-                                                        fields=layer.get('fields', {}),
-                                                        index=layer.get('index'),
-                                                        feature_count=layer.get('feature_count',None),
-                                                        configuration_options=configuration_options))
+                    upload.uploadlayer_set.add(UploadLayer(upload_file=upfile,
```

Comment:
the indent on the original code here was crazy for my taste, rather just break after (